From e9f19c6df4a6e97a36a974338196a5a0bda079e4 Mon Sep 17 00:00:00 2001
From: Victor Stinner <vstinner@redhat.com>
Date: Mon, 29 Jun 2015 16:03:56 +0200
Subject: [PATCH] Port eventlet.green.OpenSSL to Python 3

* Fix import syntax for Python 3: use relative imports
* Add unit test
---
 eventlet/green/OpenSSL/__init__.py | 10 +++++-----
 tests/openssl_test.py              | 17 +++++++++++++++++
 2 files changed, 22 insertions(+), 5 deletions(-)
 create mode 100644 tests/openssl_test.py

diff --git a/eventlet/green/OpenSSL/__init__.py b/eventlet/green/OpenSSL/__init__.py
index 56bfb8a..26b60d9 100644
--- a/eventlet/green/OpenSSL/__init__.py
+++ b/eventlet/green/OpenSSL/__init__.py
@@ -1,5 +1,5 @@
-import rand
-import crypto
-import SSL
-import tsafe
-from version import __version__
+from . import rand
+from . import crypto
+from . import SSL
+from . import tsafe
+from .version import __version__
diff --git a/tests/openssl_test.py b/tests/openssl_test.py
new file mode 100644
index 0000000..a4498dc
--- /dev/null
+++ b/tests/openssl_test.py
@@ -0,0 +1,17 @@
+import tests
+
+
+def test_import():
+    # https://github.com/eventlet/eventlet/issues/238
+    # Ensure that it's possible to import eventlet.green.OpenSSL.
+    # Most basic test to check Python 3 compatibility.
+    try:
+        import OpenSSL
+    except ImportError:
+        raise tests.SkipTest('need pyopenssl')
+
+    import eventlet.green.OpenSSL.SSL
+    import eventlet.green.OpenSSL.crypto
+    import eventlet.green.OpenSSL.rand
+    import eventlet.green.OpenSSL.tsafe
+    import eventlet.green.OpenSSL.version
